### ISSUE TITLE
High number of disk writes with 4.8.2

### ISSUE THREAD

**Issue by markuspersson** [amplitude/Amplitude-iOS#215]
I integrated the Amplitude SDK (4.8.2) in our iOS app a few weeks ago. 

Now when I follow up on some of our performance metrics I noticed quite a dramatic change in the number of disk writes. After some investigation and running the File Activity instruments I noticed that it is the Amplitude SDK that is causing the issue.

<img width="835" alt="Screenshot 2019-11-18 at 21 12 18" src="https://user-images.githubusercontent.com/1014075/69091819-6154e500-0a4b-11ea-9f8f-1bfc80c4735b.png">

<img width="906" alt="Screenshot 2019-11-18 at 21 26 18" src="https://user-images.githubusercontent.com/1014075/69091834-6d40a700-0a4b-11ea-8255-49ee1e310471.png">

<img width="1042" alt="Screenshot 2019-11-18 at 21 26 49" src="https://user-images.githubusercontent.com/1014075/69091844-716cc480-0a4b-11ea-9443-df78abda0107.png">

It is not a lot of data (80MB in total) but the difference is that we doubled the amount when adding Amplitude. 

The screenshot with excessive writes is over a time span of 1 second. So it is quite a lot of writes during the lifetime of the application. 

It would be nice if the disk writes could be batched, if they are not already.
Is there anything else I could try to set on my side to make the writes happen less frequently?

**Comment by haoliu-amp** [comment]
May I know what is the previous version you used before 4.8.2?

**Comment by markuspersson** [comment]
Sorry if I was misleading. I noticed I make it sound like there's was only an issue with 4.8.2. My intention was to only mention which version we are using.

The only version we have used in our released app is 4.8.2.

**Comment by haoliu-amp** [comment]
oh, ok, if I understand it correctly, this's your first time to use Amplitude iOS SDK, and the version is 4.8.2. Before you never used it, right?

**Comment by markuspersson** [comment]
Exactly!

We integrated Amplitude two releases ago. If you look at the first graph with the disk writes. That is when it, almost, doubled in writes.

**Comment by markuspersson** [comment]
The size itself is not the big issue for me. It is more that it seems to access the file system more than necessary. That is why I would like some sort of batching or similar. Or if there's anything I can do from the app perspective.

**Comment by haoliu-amp** [comment]
Do you call `logEvent` a lot? Any concerns over more writes? Does it slow your app?

**Comment by markuspersson** [comment]
>Do you call logEvent a lot? 

No. Currently we have a total of 15 events. One of these can be called more than once per view, but no more than 3. Most are called only once per view and some only on specific interaction.

>Any concerns over more writes?

Some writes needs to happen so I assumed it would go up, but not doubled. I have already mentioned more about my concerns previously but I can also mention that we have other analytics tools that tracks exactly the same amount but without the disk writes change as much as now.

>Does it slow your app?

Any writes has some impact on our performance. By running instruments Apple suggests which writes that could be improved.

Again, the amount of data is not my biggest concern, but doing almost 100 writes each second is.

**Comment by haoliu-amp** [comment]
Hi Markus,

Each column in the picture you shared, is disk writes for one app per day, right? 

You mentioned that there're other analytics tools has less writes. Can you give me an example? It will be good if they're also open sourced. Then we can take a look.

Thanks

**Comment by markuspersson** [comment]
No. The disk writes is for each released version of the app to App Sore; 1.0, 2.0, 3.0, e. tc.

We have a couple of less popular ones but the bigger are: Google analytics and Firebase analytics.
These have been implemented throughout each of the released versions without us seeing any high impact on disk writes. 

Worth to mention is that I'm not the one to say that they actually do less disk writes or if the have some other way of accessing the file system. We haven't noticed anything in our app at least.

**Comment by haoliu-amp** [comment]
Ok, I will spend some time to investigate. Might put this request under `enhancement` later.

**Comment by markuspersson** [comment]
Thanks! Let me know if there’s anything else I can do to help.
